esp/bindings/http/platform/httptransport.cpp

```diff
@@ -1564,11 +1564,16 @@ void CHttpRequest::getBasicAuthorization(StringBuffer& userid, StringBuffer& pas
 
     StringBuffer uidpair;
     Utils::base64decode(authheader.length() - strlen("Basic "), authheader.str() + strlen("Basic "), uidpair);
+
+    //uidpair formatted as   [domain\]username:password    (domain optional)
     
+    //Look for optional domain name separator. Assumes username cannot contain '\'
     const char* pairstr = strchr(uidpair.str(), '\\');
+    if (pairstr && 0 == strchr(pairstr,':'))//Ensure domain name separator was not part of password. Assumes username cannot contain ':'
```

Comment:
It works as desired in the case you specify, since pairstr will point to the domain backslash, and the search for the password separator begins at that backslash. However, it will break if there is both a backslash and colon in the password itself so will refactor for that